libraries/SRV_Channel/SRV_Channel_aux.cpp

```diff
@@ -211,23 +211,41 @@ void SRV_Channels::update_aux_servo_function(void)
     if (!channels) {
         return;
     }
-    function_mask.clearall();
 
-    for (uint16_t i = 0; i < SRV_Channel::k_nr_aux_servo_functions; i++) {
-        functions[i].channel_mask = 0;
-    }
-    invalid_mask = 0;
-
-    // set auxiliary ranges
+    // We build up new masks and set them at the end so that they are never
+    // in the middle of rebuilding if another thread tries to read them.
+    static Bitmask<SRV_Channel::k_nr_aux_servo_functions> new_function_mask;
+    new_function_mask.clearall();
```

Comment:
```suggestion
    static Bitmask<SRV_Channel::k_nr_aux_servo_functions> new_function_mask {};
```

Reply:
Ah, I didn't spot the static. In general i'm not a fan of statics in function for that reason. I don't have a feel for which would give better performance.